app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/FwAppInstallerActivity.java

```diff
@@ -165,13 +191,37 @@ public void onClick(View v) {
                 setInstallEnabled(false);
                 installHandler.onStartInstall(device);
                 GBApplication.deviceService().onInstallApp(uri);
+                mProgressBar.setVisibility(View.VISIBLE);
+                IntentFilter intentFilter = new IntentFilter();
+                intentFilter.addAction("nodomain.freeyourgadget.gadgetbridge.INSTALLATION_FAILED");
+                intentFilter.addAction("nodomain.freeyourgadget.gadgetbridge.INSTALLATION_SUCCESS");
+                registerReceiver(installationResultReceiver, intentFilter);
             }
         });
 
         uri = getIntent().getData();
         if (uri == null) { //for "share" intent
             uri = getIntent().getParcelableExtra(Intent.EXTRA_STREAM);
         }
+
+        //Get apps count and app index from calling intent
+        int appsCount = getIntent().getIntExtra("APPS_COUNT", 1);
+        int currentAppIndex = getIntent().getIntExtra("APP_INDEX", 1);
+
+        //Set up the toolbar
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        toolbar.setTitleTextColor(ContextCompat.getColor(getApplicationContext(), android.R.color.white));
```

Comment:
I agree, we didn't because of lack of time on our end, perhaps you can open a new PR after this one is merged to take care of the migration? It would be really great!